pandas/tests/test_indexing.py

```diff
@@ -5619,19 +5650,12 @@ def f():
                           slice(3, 4.0),
                           slice(3.0, 4.0)]:
 
-                    def f():
-                        s.iloc[l]
-                    self.assertRaises(TypeError, f)
-
-                    def f():
-                        s.loc[l]
-                    self.assertRaises(TypeError, f)
+                    for idxr in [lambda x: x.loc,
+                                 lambda x: x.iloc,
+                                 lambda x: x]:
+                        self.assertRaises(TypeError, lambda: idxr(s)[l])
```

Comment:
https://github.com/pydata/pandas/issues/12333#issuecomment-185246550

this is the same as your comment issue. We don't allow float indexers in slicing at ALL, except for ``Float64Index``. This is possible to relax, but introduces another layer of complexity here. (this doesn't apply for ``.ix``, just ``[]`` and ``.loc``)